### PR TITLE
Oozelings no longer autocannibalize inorganic limbs

### LINKED ISSUE #9988: [Bug]: Prostetic limbs are absorbed on low blood for oozelings who take prostetic limb as character trait
### BYOND Version

_No response_

### Round Date

2023/10/07

### Round ID

46148

### Test merges

_No response_

### Describe the bug

When I did lost all blood as an oozeling with a robotic limb, I noticed the limb retract into my body, to be replaced with a normal oozeling arm upon regeneration.
Additional notes: Death was caused by heretic blade, it is unknown if it was Mark of flesh which makes targets bleed Profusely.
Related media:
https://cdn.discordapp.com/attachments/568515045235884065/1160384693619920916/image.png?ex=65347775&is=65220275&hm=c6c94497bce85aeb81efed23686bc38d10033e2de100bc8e0f09a41f1db47389&

### What did you expect?

I expected the Robot arm to remain while at low blood

### Reproducible steps

1. In player preferences, set race to oozeling, and add prosthetic limb quirk
2. Launch game
3. Proceed to loose all blood
4. Observe robotic limb retract into body

### PR DESCRIPTION
**Pull request by VickiMorris**
## About The Pull Request

Oozelings that have low enough blood to start autocannibalizing/reabsorbing limbs will no longer absorb limbs that are flagged as inorganic (such as prosthetics). Otherwise, they still follow the same behavior of cannibalizing legs, followed by arms, and then causing breathloss if there are no limbs to consume.

Adds an explicit return type to get_bodypart() because it was complaining about field access while running IS_ORGANIC_LIMB() while writing this change.

## Why It's Good For The Game

While it is funny conceptually, and horrifying on an in-character note to watch an oozeling absorb and directly convert a hunk of metal into more blood and food, it doesn't really make sense (especially with how the description of reabsorbing limbs is written).

Closes #9988

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/410916fe-2dc5-4f75-b0da-9a9ca8bd9a1a

</details>

## Changelog
:cl: Impish_Delights
fix: Oozelings no longer autocannibalize inorganic limbs while at low blood
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm [hunk 1]
```diff
@@ -104,11 +104,15 @@
 /datum/species/oozeling/proc/Cannibalize_Body(mob/living/carbon/human/H)
 	var/list/limbs_to_consume = list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG) - H.get_missing_limbs()
 	var/obj/item/bodypart/consumed_limb
+	var/L
+	for(L in limbs_to_consume) //Check every bodypart the oozeling has, see if they're organic or not
+		if(!IS_ORGANIC_LIMB(H.get_bodypart(L))) //Get actual limb, list only has body zone
+			limbs_to_consume -= list(L) //If it's inorganic, remove it from the consumption list
```
Comment (Penwin0) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Doesn't need to be a list does it?

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm [hunk 2]
```diff
@@ -104,11 +104,15 @@
 /datum/species/oozeling/proc/Cannibalize_Body(mob/living/carbon/human/H)
 	var/list/limbs_to_consume = list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG) - H.get_missing_limbs()
 	var/obj/item/bodypart/consumed_limb
+	var/L
```
Comment (Penwin0) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Declaring the variable inside the for is the norm

Comment (Penwin0) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Like for (var/L ......)

**Comment by Rukofamicom** [comment]
Ideally oozelings should be completely incompatible with inorganic limbs tbh

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm (hunk 1 above)
Comment (VickiMorris) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: It does not, thank you for pointing this out.

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm (hunk 2 above)
Comment (VickiMorris) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Changed it to declare the var inside the for() loop. Did it the other way as that's how the DM reference guide shows it (probably for ease of reading/teaching).

**Comment by VickiMorris** [comment]
I have the hilarious image of a doctor treating low blood on an oozeling by giving them ligament hooks, then repeatedly attaching prothestic limbs that they keep absorbing.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**